plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -110,7 +110,9 @@ public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
         executorBuilders.add(new ScalingExecutorBuilder(PRIORITY_STREAM_READER, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5)));
 
         executorBuilders.add(new FixedExecutorBuilder(settings, FUTURE_COMPLETION, normalPoolCount(settings), 10_000, FUTURE_COMPLETION));
-        executorBuilders.add(new ScalingExecutorBuilder(STREAM_READER, 1, halfProcMaxAt5, TimeValue.timeValueMinutes(5)));
+        executorBuilders.add(
+            new ScalingExecutorBuilder(STREAM_READER, 1, 3 * allocatedProcessors(settings), TimeValue.timeValueMinutes(5))
```

Comment:
Add some comment here as well.

Reply:
This is just an increase in capacity. What do you want me to put?

Reply:
Explain why you think 3*allocatedProcessors is good?

Reply:
The explanation is that there are disk reads i.e IO happening in stream reader pool. This explanation will not hold true with the optimization planned https://github.com/opensearch-project/OpenSearch/issues/11263